### PR TITLE
Delete (& backup) invalid NPC accounts created by poor econ code

### PR DESCRIPTION
**Pull request by JRoy**
my magnum opus

### CODE REVIEW

**Comment by JRoy** [comment]
merge #4484 first

File: Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java [hunk 1]
```diff
@@ -243,6 +250,112 @@ public void convertStupidCamelCaseUserdataKeys() {
...
+
+                try {
+                    //noinspection UnstableApiUsage
+                    Files.move(file, new File(backupFolder, file.getName()));
+                    movedAccounts.incrementAndGet();
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, "Error while moving NPC file", e);
+                }
+            } catch (final RuntimeException ex) {
+                LOGGER.log(Level.INFO, "File: " + file);
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java:364]:
```suggestion
                LOGGER.log(Level.INFO, "File: " + file);
                feedbackTask.cancel(false);
                executor.shutdown();
```

File: Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java [hunk 2]
```diff
@@ -243,6 +250,112 @@ public void convertStupidCamelCaseUserdataKeys() {
...
+                throw ex;
+            }
+        }
+        feedbackTask.cancel(false);
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java:370]:
```suggestion
        feedbackTask.cancel(false);
        executor.shutdown();
```

File: Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java [hunk 3]
```diff
@@ -243,6 +250,134 @@ public void convertStupidCamelCaseUserdataKeys() {
...
+        LOGGER.info(" skip the process until you rename or remove the backup folder.");
+        LOGGER.info("#===========================================================================#");
+
+        final int totalUserFiles = userFiles.length;
+        LOGGER.info("Found ~" + totalUserFiles + " files under \"plugins/Essentials/userdata\"...");
+
+        final AtomicInteger movedAccounts = new AtomicInteger(0);
+        final AtomicInteger totalAccounts = new AtomicInteger(0);
+
+        // Less spammy feedback for greater userdata counts: 100 files -> 5 seconds, 1k -> 7s, 10k -> 9s, 100k -> 11s, 1m -> 14s
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java]:
```suggestion
        // Less spammy feedback for greater userdata counts, eg 100 files -> 5 seconds, 1k -> 7s, 10k -> 9s, 100k -> 11s, 1m -> 14s
```

**Review by mdcfe — APPROVED** [review]

**Pull request merged**